tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Serializers/KnownSerializers/KnownSerializerFinderTests.cs

```diff
@@ -0,0 +1,222 @@
+/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq.Expressions;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Driver.Linq.Linq3Implementation.Serializers.KnownSerializers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Serializers.KnownSerializers
+{
+    public class KnownSerializerFinderTests
+    {
+        #region static
+        private static IBsonDocumentSerializer GetCollectionSerializer()
+        {
+            return (IBsonDocumentSerializer)BsonSerializer.LookupSerializer<C>();
+        }
+        #endregion
+
+        private enum E { A, B }
+
+        private class C
+        {
+            public int P { get; set; }
+            [BsonRepresentation(BsonType.Int32)]
+            public E Ei { get; set; }
+            [BsonRepresentation(BsonType.String)]
+            public E Es { get; set; }
+            public A A { get; set; }
+        }
+
+        private class A
+        {
+            public B B { get; set; }
+        }
+
+        private class B { }
+
+        private class View
+        {
+            public A A { get; set; }
+        }
+
+        [Fact]
+        public void Identity_expression_should_return_collection_serializer()
+        {
+            Expression<Func<C, C>> expression = x => x;
+            var collectionSerializer = GetCollectionSerializer();
+
+            var result = KnownSerializerFinder.FindKnownSerializers(expression, collectionSerializer);
+
+            var serializer = result.GetSerializer(expression.Body);
+            serializer.Should().Be(collectionSerializer);
+        }
+
+        [Fact]
+        public void Int_property_expression_should_return_int_serializer()
+        {
+            Expression<Func<C, int>> expression = x => x.P;
+            var collectionSerializer = GetCollectionSerializer();
+
+            var result = KnownSerializerFinder.FindKnownSerializers(expression, collectionSerializer);
+
+            var serializer = result.GetSerializer(expression.Body);
+            collectionSerializer.TryGetMemberSerializationInfo(nameof(C.P), out var expectedPropertySerializer).Should().BeTrue();
```

Comment:
Rename out var to `expectedPropertySerializationInfo`.

Similarly everywhere else.

Reply:
I meant to do this since the variables represent a serializationInfo, not a serializer. Thanks for reminding me.